src/jvm/main/org/jetbrains/lincheck/Lincheck.kt

```diff
@@ -79,9 +80,15 @@ object Lincheck {
             val verifier = testCfg.createVerifier()
             val wrapperClass = GeneralPurposeModelCheckingWrapper::class.java
             testCfg.createStrategy(wrapperClass, scenario, null, null).use { strategy ->
+                check(strategy is ModelCheckingStrategy)
+
+                // save fields referenced by lambda for restoring by the snapshot tracker
+                block.javaClass.declaredFields
+                    .mapNotNull { readFieldSafely(block, it).getOrNull() }
+                    .forEach(strategy::updateSnapshotWithRootObject)
```

Comment:
Probably worth moving this code to some other place, e.g., to `ManagedStrategy` constructor/initialization, to make sure it is called from all places that run `ManagedStrategy` over the lambda. 

Because, for instance, `runPluginReplay` currently also creates an instance of the `ManagedStrategy`, and there you would not add the fields to the snapshot.

Reply:
done